notebooks/2023-01-23 Differential abundance testing.Rmd

```diff
@@ -57,34 +53,78 @@ table(rowSums(is.na(prot_res_flt)))
+  geom_hline(yintercept=log2(1/1), linetype=2)
 ```
+There is a lot of variation in the distribution of ratios between the experiment, with the even replicates (CL=L) having consistently higher ratios (aside from replicate 16). Even within one half of the label swap, there is a lot of variation. We can separate the sample by their annotations to explore this futher (see below). The green line is the expected maximum value for a true RBP and the black line is the expected minimum value for a non RBP. However, values can fall outside these bounds due to measurement error, incomplete incorporation and/or unbalanced cell numbers for each isotope.
+
 
 ```{r}
-#need to 2^ value in order to un-log the value
-#need to -0.5 the ratio.
-#in this chunk i attempt to change the value column from log values back to the ratio, and i attempt to -0.5 from the value.
-  
+
+prot_res_flt_long <- tidy(prot_res_flt, addPheno =TRUE) %>%
+  merge(fData(prot_res_flt),by.x='protein', by.y='row.names')
+
+p <- prot_res_flt_long %>%
+  filter(go_rbp_glyco!='GO:RBP & glycoprotein') %>%
+  ggplot(
+            aes(factor(Replicate, levels=1:16), value, colour=go_rbp_glyco, group=interaction(go_rbp_glyco, Replicate))) +
+  geom_boxplot(outlier.shape = NA) +
+  scale_colour_manual(values=c('grey', 'grey20', get_cat_palette(3)[c(3,2)])) +
+  facet_wrap(~label_swap, scales='free') +
+  theme_camprot(base_size=10, base_family='sans', aspect_square=FALSE, border=FALSE) +
+  geom_hline(yintercept=log2(1/1), linetype=2, colour=get_cat_palette(3)[3]) +
+  geom_hline(yintercept=log2(1/2), linetype=2, colour='grey20') +
+  theme(strip.background=element_blank()) +
+  ylab('CL/NC ratio') +
+  xlab('Replicate') +
+  coord_flip(ylim=c(-2.5, 2.5))
   
-#Here i attempt to raise the ratio to the power of 10 (opposite of log)
-#should be opposite of log2 which is 2^value
-silac_unlog <- silac_protein_tidy %>%
-  mutate(silac_protein_tidy, value= 2^value)
 
-#Here i attempt to subtract 0.5
-silac_0.5 <- 
-   mutate(silac_unlog, value=value-0.5)
-#this just made things a bit easier 
-silac_protein_tidy <- silac_0.5
+print(p)
+```
+Now that we have included the annotation, we can see that althought there is a large variation in the ratio distributions between the experiments, in almost all experiments, the GO:RBPs have higher ratios.
 
+Since glycoproteins should not have higher abundance upon CL and should have their CL/NC ratio centered at 0, we can use this prior expectation to adjust the ratio. Following this adjustment, our null hypothesis in the downstream statistical testing is that the protein has the same CL/NC enrichment as a typical glycoprotein, which should not be enriched by CL.
 
+See https://mrctoxbioinformatics.github.io/Proteomics_data_analysis/Markdowns/LFQ_alternative_normalisation.html for code to do this ratio adjustment. Once the ratios have been adjusted, we can proceed with the statistical testing as previously.
 
+```{r}
+# insert code for adjustment here...
```

Comment:
This is where you can include the neccessary code to adjust the ratios with respect to the glycoproteins.